pkg/plugins/runtime/k8s/controllers/pod_converter.go

```diff
@@ -192,46 +192,14 @@ func GatewayByServiceFor(clusterName string, pod *kube_core.Pod, services []*kub
 	}, nil
 }
 
-// DeploymentFor returns the name of the deployment that the pod exists within. The second return
-// value indicates whether or not the deployment was found when no error occurs, otherwise an
-// error is returned as the third return value.
-func (p *PodConverter) DeploymentFor(ctx context.Context, namespace string, pod *kube_core.Pod) (string, bool, error) {
-	owners := pod.GetObjectMeta().GetOwnerReferences()
-	var rs *kube_apps.ReplicaSet
-	for _, owner := range owners {
-		if owner.Kind == "ReplicaSet" {
-			rs = &kube_apps.ReplicaSet{}
-			rsKey := kube_client.ObjectKey{Namespace: namespace, Name: owner.Name}
-			if err := p.ReplicaSetGetter.Get(ctx, rsKey, rs); err != nil {
-				return "", false, err
-			}
-			break
-		}
-	}
-
-	if rs == nil {
-		return "", false, nil
-	}
-
-	rsOwners := rs.GetObjectMeta().GetOwnerReferences()
-	for _, owner := range rsOwners {
-		if owner.Kind == "Deployment" {
-			return owner.Name, true, nil
-		}
-	}
-
-	return "", false, nil
-}
-
 func (p *PodConverter) GatewayByDeploymentFor(ctx context.Context, clusterName string, pod *kube_core.Pod, services []*kube_core.Service) (*mesh_proto.Dataplane_Networking_Gateway, error) {
 	namespace := pod.GetObjectMeta().GetNamespace()
-	deployment, found, err := p.DeploymentFor(ctx, namespace, pod)
+	deployment, kind, err := p.InboundConverter.NameExtractor.Name(ctx, pod)
 	if err != nil {
 		return nil, err
 	}
-	if !found {
-		// Fall back on old service tags if Pod not part of Deployment
-		return GatewayByServiceFor(clusterName, pod, services)
+	if !strings.Contains(kind, "Deployment") {
```

Comment:
should be `==` probably

Reply:
I think for correctness we should be dealing with `GroupKinds` all throughout here but...